tests/test_causal_graph.py

```diff
@@ -1412,3 +1412,13 @@ def test_node_repr(self):
         self.assertEqual(tscg['b'], tscg.get_node('b'))
         self.assertEqual(repr(tscg['a']), 'TimeSeriesNode("a")')
         self.assertEqual(repr(tscg['b']), 'TimeSeriesNode("b", type="continuous")')
+
+    def test_add_to_undirected_edges(self):
+
+        cg = CausalGraph()
+        cg.add_edge('a', 'b', edge_type=EdgeType.UNDIRECTED_EDGE)
+
+        with self.assertRaises(CausalGraphErrors.ReverseEdgeExistsError):
+            cg.add_edge('b', 'a')
+        with self.assertRaises(CausalGraphErrors.ReverseEdgeExistsError):
+            cg.add_edge('b', 'a', edge_type=EdgeType.UNDIRECTED_EDGE)
```

Comment:
Could you also add assertions like above that check for other edge types? You can check out the `EdgeType` Enum to see what other edge types there are

Reply:
Do you mean other edge types for the first edge `cg.add_edge('a', 'b', edge_type=EdgeType.UNDIRECTED_EDGE)` or in adding edge b -> a?

Reply:
The former, i.e. `EdgeType.BIDIRECTED_EDGE`, etc. Your fix should work regardless, but it's good practice to protect against future changes on the backend